browser/brave_wallet/brave_wallet_service_unittest.cc

```diff
@@ -315,6 +316,12 @@ class BraveWalletServiceUnitTest : public testing::Test {
     }
 #endif
 
+    brave_wallet::BraveWalletServiceFactory::GetInstance()
```

Comment:
why the change here to use the factory for the unit test? That seems to make it more complicated than necessary?

Reply:
I think the reason was that the tests check some histogram values and with one service already running and another being created in the test the values were off. I suppose we could adjust the histogram expectations instead?

Reply:
do you know what the exact error was?

Reply:
I guess it makes sense that we wouldn't want to create the service twice for the same profile, but I think we can do this with `BrowserContextKeyedServiceFactory::SetTestingFactory`

Reply:
Changed in 1ec669ebfc2. Turns out that `TestingProfile::Builder` has a method to add a testing factory.